src/index.ts

```diff
@@ -79,6 +62,42 @@ const discordUserAnnouncementDictionary: { [key: string]: string } = {
   'NutDragSwag#0374': 'alexander_the_great.mp3'
 };
 
+// The VoiceBasedChannel the bot is connected to (null if not connected)
+let channel: VoiceBasedChannel | null | undefined = null;
+// The interval ID for polling the League of Legends Live Game Client API (null if not polling)
+let leagueOfLegendsPollIntervalId: NodeJS.Timer | null = null;
+
+// Event triggered when there is an error with the audio player
+audioPlayer.on('error', (error) => {
+  console.log('Error with audio player:', error);
+});
+
+// Event triggered when the client becomes ready to start working
+client.once('ready', async (client) => {
+  // Get the guild this bot is in
+  const guild = await client.guilds.fetch(GUILD_ID);
+  // If the bot starts in a VoiceChannel, find out which channel it is
+  guild.channels.cache.forEach((curChannel: GuildBasedChannel) => {
+    if (curChannel.isVoiceBased() && curChannel.members.has(client.user.id)) {
+      // Set the channel this bot is in
+      channel = curChannel;
+      // If a member in this channel is in the middle of a LoL game and the announcer is enabled, start polling
+      curChannel.members.forEach((member: GuildMember) => {
+        if (
+          member.presence &&
+          presenceIndicatesPlayingLeagueOfLegends(member.presence) &&
+          member.presence.activities[0].state === PresenceState.IN_GAME &&
+          IS_LOL_ANNOUNCER_ENABLED
+        ) {
+          console.log('Polling game...');
+          leagueOfLegendsPollIntervalId = pollCurrentGame(curChannel, audioPlayer, PATH_TO_CLIPS);
```

Comment:
For consistency with elsewhere (and since `curChannel` and `channel` should be the same value here, maybe we should pass `channel` to `pollCurrentGame` instead

Reply:
Not sure why but it was having trouble seeing `channel` as a `VoiceBasedChannel`, so I would've had to do `channel as VoiceBasedChannel` instead. Since you told me that was bad practice I opted away from it. I could've added a check for `channel` in the `if` statement too, but that seemed unnecessary because we're setting it right above. `curChannel` worked just fine, that's why I'm feeding that in here

Reply:
I'm good with merging as is for now if you are. Thanks for looking into it!